File: Common/InMotionGIT.Common/DataAcceess/WhereStatements/StringCondition.cs

namespace InMotionGIT.Common.DataAccess.WhereStatements
{

    public class StringCondition<t>
    {

        private readonly string fieldName;
        private readonly IWhereStatement tableType;

        public StringCondition(string name, IWhereStatement @base)
        {
            fieldName = name;
            tableType = @base;
        }

        public LogicalOperator<t> EqualTo(string value)
        {
            tableType.command += string.Format("{0} = '{1}'", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> NotEqualTo(string value)
        {
            tableType.command += string.Format("{0} <> '{1}'", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> Like(string value)
        {
            tableType.command += string.Format("{0} LIKE '{1}'", fieldName, value);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> IsNull()
        {
            tableType.command += string.Format("{0} IS NULL", fieldName);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> IsNotNull()
        {
            tableType.command += string.Format("{0} IS NOT NULL", fieldName);
            return new LogicalOperator<t>(tableType);
        }

        // StartsWith
        // EndsWith

    }

}

File: Common/InMotionGIT.Common/DataAcceess/WhereStatements/StringCondition.vb
Namespace DataAccess.WhereStatements

    Public Class StringCondition(Of t)

        Private ReadOnly fieldName As String
        Private ReadOnly tableType As IWhereStatement

        Public Sub New(name As String, base As IWhereStatement)
            fieldName = name
            tableType = base
        End Sub

        Public Function EqualTo(value As String) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} = '{1}'", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function NotEqualTo(value As String) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} <> '{1}'", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function [Like](value As String) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} LIKE '{1}'", fieldName, value)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function IsNull() As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} IS NULL", fieldName)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function IsNotNull() As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} IS NOT NULL", fieldName)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        'StartsWith
        'EndsWith

    End Class

End Namespace